File: Iteration_2/src/main/java/StudentAffairsStaffInterface.java
import java.util.ArrayList;
import java.util.InputMismatchException;
import java.util.Scanner;

public class StudentAffairsStaffInterface implements UserInterface {
        StudentAffairsStaff staff;
        Scanner scan;
        ArrayList<Course> courses;
        ArrayList<CourseSection> courseSections;
        
    
        public StudentAffairsStaffInterface() {
            scan = new Scanner(System.in);
           

        }
    
        public StudentAffairsStaffInterface(StudentAffairsStaff staff, ArrayList<Course> courses, ArrayList<CourseSection> courseSections) {
            this.staff = staff;
            this.courses = courses;
            this.courseSections = courseSections;
            scan = new Scanner(System.in);
        }

    
        public boolean showMenu() {
            switch (getChoice()) {
                case 1:
                    addCourse();
                    break;
                case 2:
                    removeCourse();
                    break;
                case 3:
                    System.out.println("You have successfully logged out\n");
                    return true;
                default:
                    System.out.println("Enter 1, 2 or 3.");
                    break;
            }
    
            return false;

        }
    
        public int getChoice() {
            System.out.print("1-  Add Course\n2-  Remove Course\n3-  Log Out\nSelect an operation: ");
            int choice = 0;
            try {
                choice = scan.nextInt();
            } catch (InputMismatchException e) {
                System.out.println("Enter an integer value.");
            } catch (Exception e) {
                System.out.println("There is an error in getChoice method.");
            }
            return choice;
        }
    
        public void addCourse() {
            String[] course = askCourseParameters();
            try {
                Course newCourse = staff.createCourse(course[0], course[1], course[3],Integer.parseInt(course[2]), Integer.parseInt(course[4])); 
                courses.add(newCourse);
                JsonManagement.getInstance().writeCourseToJson();
                courseSections.addAll(newCourse.getCourseSections());
                
            } catch (NumberFormatException e) {
                System.out.println("Enter an integer value for course code and course credits.");
            }
        }
    
     


    public String[] askCourseParameters() {
        String[] course = new String[5];
        try {
            System.out.print("Enter course name: ");
            course[0] = scan.next();
            System.out.print("Enter course code: ");
            course[1] = scan.next();
            System.out.print("Enter course credits: ");
            course[2] = scan.next();
            System.out.print("Enter course type (Mandatory (m) / Technical Elective (te) / Non-Technical Elective (nte) ):  ");
            course[3] = scan.next().toLowerCase();
            if(!(course[3].equals("m") || course[3].equals("te") || course[3].equals("nte")) )
             throw new InputMismatchException("Invalid choice! ");
            System.out.print("How many sections does the course have? : ");
            course[4] = scan.next();


        } catch (InputMismatchException e) {
            System.out.println(e.getMessage() + " Please enter a valid choice!");
        } catch (Exception e) {
            System.out.println("There is an error in askCourseParameters method.");
        }
        return course;
    }

   

    public void removeCourse() {
        printCourses();
        System.out.println("Which course do you want to remove?");
        int courseIndex = -1;
        Course removedCourse = null;
        try {
            courseIndex = scan.nextInt() - 1;
            removedCourse = courses.remove(courseIndex);
        } catch (InputMismatchException e) {
            System.out.println("Enter an integer value.");
            return;
        }
        System.out.println("The " + removedCourse.getCourseId() + " course has been removed.");
    }

    public void printCourses() {
        for (int i = 0; i < courses.size(); i++) {
            System.out.println(i + 1 + "- " + courses.get(i).toString());
        }
    }
}

File: Iteration_3/StudentAffairsStaffInterface.py
import sys
from typing import List
from CourseSection import CourseSection
from Course import Course

class StudentAffairsStaffInterface:
    def __init__(self, staff, courses: List['Course'], course_sections: List['CourseSection']):
        self.staff = staff
        self.courses = courses
        self.course_sections = course_sections

    def show_menu(self):
        while True:
            choice = self.get_choice()
            if choice == 1:
                self.add_course()
            elif choice == 2:
                print("You have successfully logged out\n")
                return True
            else:
                print("Enter 1 or 2.")

    def get_choice(self):
        print("1-  Add Course\n2-  Log Out\nSelect an operation: ", end="")
        try:
            choice = int(input())
        except ValueError:
            print("Enter an integer value.")
            choice = -1
        return choice

    def add_course(self):
        course_params = self.ask_course_parameters()
        try:
            new_course = self.staff.create_course(
                course_params[0], course_params[1], course_params[3], int(course_params[2]), int(course_params[4])
            )
            self.courses.append(new_course)
            self.course_sections.extend(new_course.get_course_sections())
            # JsonManagement.getInstance().write_course_to_json()  # Uncomment when implementing JSON handling
        except ValueError:
            print("Enter an integer value for course code and course credits.")

    def ask_course_parameters(self):
        course = [None] * 5
        try:
            course[0] = input("Enter course name: ")
            course[1] = input("Enter course code: ")
            course[2] = input("Enter course credits: ")
            course[3] = input("Enter course type (Mandatory (m) / Technical Elective (te) / Non-Technical Elective (nte)): ").lower()

            if course[3] not in ["m", "te", "nte"]:
                raise ValueError("Invalid choice!")

            course[4] = input("How many sections does the course have?: ")

        except ValueError as e:
            print(f"{e} Please enter a valid choice!")
        return course

    def remove_course(self):
        self.print_courses()
        print("Which course do you want to remove?")
        try:
            course_index = int(input()) - 1
            removed_course = self.courses.pop(course_index)
            print(f"The {removed_course.get_course_id()} course has been removed.")
        except ValueError:
            print("Enter an integer value.")
        except IndexError:
            print("Invalid course selection.")

    def print_courses(self):
        for i, course in enumerate(self.courses, 1):
            print(f"{i}- {course}")
